src/dotnet/MulticoreJitProfilePathCalculator.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.Extensions.PlatformAbstractions;
+
+namespace Microsoft.DotNet.Cli
+{
+    internal class MulticoreJitProfilePathCalculator
+    {
+        private string _multicoreJitProfilePath;
+
+        public string MulticoreJitProfilePath
+        {
+            get
+            {
+                if (_multicoreJitProfilePath == null)
+                {
+                    CalculateProfileRootPath();
+                }
+
+                return _multicoreJitProfilePath;
+            }
+        }
+
+        private void CalculateProfileRootPath()
+        {
+            var profileRoot = GetRuntimeDataRootPath();
+
+            var version = Product.Version;
+
+            var rid = PlatformServices.Default.Runtime.GetRuntimeIdentifier();
+
+            _multicoreJitProfilePath = Path.Combine(profileRoot, "sdk", version, rid, "optimizationdata");
+        }
+
+        private string GetRuntimeDataRootPath()
+        {
+            return RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
+                ? GetWindowsRuntimeDataRoot()
+                : GetNonWindowsRuntimeDataRoot();
+        }
+
+        private static string GetWindowsRuntimeDataRoot()
```

Comment:
Should these locations be overridable with an environment variable?  That way if someone wants to move them (like in a docker scenario), they can.  Say they are on a daemon in unix and there is no HOME dir, or the process doesn't have access to it.  But they still want the multi-core JIT behavior.